ampd/src/grpc/reqs.rs

```diff
@@ -24,12 +26,40 @@ pub fn validate_broadcast(req: Request<BroadcastRequest>) -> Result<Any, Error>
         .ok_or(report!(Error::EmptyBroadcastMsg))
 }
 
+pub fn validate_smart_contract_state(
+    req: Request<SmartContractStateRequest>,
+) -> Result<(TMAddress, Vec<u8>), Error> {
+    let SmartContractStateRequest { contract, query } = req.into_inner();
+
+    ensure!(!query.is_empty(), Error::InvalidQuery);
+    let query: serde_json::Value =
+        serde_json::from_slice(&query).change_context(Error::InvalidQuery)?;
+
+    Ok((
+        validate_address(&contract)?,
+        serde_json::to_vec(&query).expect("must be valid JSON"),
```

Comment:
is there a case where you can deserialize it but not serialize it again? Why not take the original value?